src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/impl/ValidationServiceImpl.java

```diff
@@ -114,6 +121,50 @@ public ValidateRulesResult validateRules(SwitchId switchId, List<FlowEntry> pres
         return new HashSet<>();
     }
 
+    @Override
+    public ValidateGroupsResult validateGroups(SwitchId switchId, List<GroupEntry> presentGroups) {
+        Collection<MirrorGroup> expected = mirrorGroupRepository.findBySwitchId(switchId);
+        Set<Integer> expectedGroups = expected
+                .stream()
+                .map(group -> (int) group.getGroupId().getValue()).collect(Collectors.toSet());
+
+        Set<Integer> presentGroupsIds = presentGroups.stream()
+                .map(GroupEntry::getGroupId)
+                .filter(group -> 1 != group) // NOTE(tdurakov): exclude default group
+                .collect(Collectors.toSet());
+        Set<Integer> missingGroups = new HashSet<>(expectedGroups);
+        missingGroups.removeAll(presentGroupsIds);
+        if (!missingGroups.isEmpty() && log.isErrorEnabled()) {
+            log.error("On switch {} the following groups are missed: {}", switchId,
+                    missingGroups.stream().map(x -> Integer.toString(x))
+                            .collect(Collectors.joining(", ", "[", "]")));
+        }
+        Set<Integer> properGroups = new HashSet<>(expectedGroups);
```

Comment:
I think here must be `new HashSet<>(presentGroupsIds)`

because 
```
Set<Integer> properGroups = new HashSet<>(expectedGroups);
properGroups.retainAll(expectedGroups);
```
creates set of expected groups and retain same expected groups